src/OpenTelemetry/Metrics/Aggregators/DoubleLastValueAggregator.cs

```diff
@@ -22,47 +22,31 @@ namespace OpenTelemetry.Metrics.Aggregators
     /// <summary>
     /// Simple aggregator that only keeps the last value.
     /// </summary>
-    /// <typeparam name="T">Type of measure instrument.</typeparam>
-    public class LastValueAggregator<T> : Aggregator<T>
-        where T : struct
+    public class DoubleLastValueAggregator : Aggregator<double>
     {
-        private T value;
-        private T checkpoint;
-
-        public LastValueAggregator()
-        {
-            if (typeof(T) != typeof(long) && typeof(T) != typeof(double))
-            {
-                throw new Exception("Invalid Type");
-            }
-        }
+        private double value;
+        private double checkpoint;
 
         public override void Checkpoint()
         {
             this.checkpoint = this.value;
```

Comment:
Not from your change @alexeyzimarev but could you please change here and on Update to use `Interlocked.Exchange` so #585 is also fixed.

Reply:
Done